### ISSUE TITLE
Running Wireguard-UI on a local server

### ISSUE THREAD

**Issue by xiki808** [ngoduykhanh/wireguard-ui#39]
Can this run on a local server running behind a router?

I have the image running on the server. I can access the web UI on the local server's IP (192.168.1.XXX:5000) and in the configurations I see that it picked up my public IP provided by the ISP.

I create a new client and leave everything on default settings. On my android, from WireGuard app, after I scan the QRCode and enable the connection, I don't get internet anymore. I tried this from the same local Wifi and on 3G with same results.

I also tried changing the DNS to the one provided by the router. I also opened the port 51820 on the router - with External and Internal on the same port, UDP and pointing to the server's local IP ( I also have port 443 and 22 open pointing to the server's port, which I'm hoping to close soon )

I would appreciate any tips, as I'm lost and would really like to use this service!

I installed WireGuard on my local server following this guide: [here](https://nixvsevil.com/posts/wireguard-in-proxmox-lxc/) up to the part it configures an LXC, which I'm hoping to replace with your docker image.

**Comment by ngoduykhanh** [comment]
> I create a new client and leave everything on default settings. On my android, from WireGuard app, after I scan the QRCode and enable the connection, I don't get internet anymore. I tried this from the same local Wifi and on 3G with same results.

Just to clarify, were you able to ping/reach the wireguard server using its private IP address (the wireguard network IP)? For internet access, it requires NAT configuration as well. In may use case, I usually use the following iptables rules to have internet access.

```
iptables -A FORWARD -i wg0 -j ACCEPT
iptables -A FORWARD -i eth0 -j ACCEPT
iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE; ip6tables -A FORWARD -i wg0 -j ACCEPT
ip6tables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
```

Adjust the script and try it in our environment. If it works, you can apply it via Wireguard-UI by defining the `Post Up Script`  in `Wireguard Server` section.

**Comment by xiki808** [comment]
Yes, I am able to get a response when pinging the server from the client and vise-versa.

Wireguard also returns this, when sending the `wg` command on the server, where the image is loaded.

![image](https://user-images.githubusercontent.com/8024321/96476110-20ee8000-1235-11eb-9348-5601b6869827.png)

I've been playing around with iptables, but I still cannot get any internet.

I tried your suggestions, and also others I found online. I also checked this [guide](https://www.cyberciti.biz/faq/how-to-set-up-wireguard-firewall-rules-in-linux/), which provides the following configurations:

```
IPT="/sbin/iptables"

IN_FACE="eth0"                   # NIC connected to the internet
WG_FACE="wg0"                    # WG NIC 
SUB_NET="10.8.1.0/24"            # WG IPv4 sub/net aka CIDR  <--- This I removed. Not sure what to place here.
WG_PORT="51820"                  # WG udp port
 
# IPv4 rules #
$IPT -t nat -D POSTROUTING -s $SUB_NET -o $IN_FACE -j MASQUERADE
$IPT -D INPUT -i $WG_FACE -j ACCEPT
$IPT -D FORWARD -i $IN_FACE -o $WG_FACE -j ACCEPT
$IPT -D FORWARD -i $WG_FACE -o $IN_FACE -j ACCEPT
$IPT -D INPUT -i $IN_FACE -p udp --dport $WG_PORT -j ACCEPT
```

When configuring nat POSTROUTING I removed the `-s` source flag, as I was not sure what to place. Should this be the subnet configured under Interfaces in wg0.conf?

Should I open some port on my router to make this work? Currently I have incoming PORT 51820 UDP to local PORT 51820 configured on my router.

I'm ignoring IPv6 configurations at the moment, to make my testing easier, I don't think this is important for now.

**Comment by xiki808** [comment]
OK I managed to get it to work :) 

My issue was that I was using the wrong interface. When using Proxmox, you will have a Network Device with name enp2s0f0 or something, and a Linux Bridge with the name of vmbr0. 

I had to use the vmbr0 in my iptables configurations.

Further more, during my tests I had to make sure that I restart and reload everything. Sometimes the WireGuard on the server seemed to still connect to a client which I had removed. A restart to wireguard service helped fixing this everytime I changed settings and wanted to create a new client.

@ngoduykhanh Thank you, your comment helped me to focus on the iptable configurations.
